### ISSUE TITLE
4 channel data

### ISSUE THREAD

**Issue by fyang064** [sebastianvitterso/master-sau#1]
Hi, I'm wondering how to train your yolo_four_input model while preparing RGB data and another infrared data. Could you please give me some hints?

**Comment by sebastianvitterso** [comment]
Hey, Fyang.

I am not entirely sure what you are asking. 

To train the four-input model (available in [branch four-input](https://github.com/sebastianvitterso/master-sau/tree/four-input)), your folder structure should look something like: 

```
slurm/...
yolov5/...
data/
  train/
    images/
    ir/
    labels/
  validation/
    images/
    ir/
    labels/
  test/
    images/
    ir/
    labels/
```

Additionally, your `images`, `ir` and `labels`-folders must contain files with similar names:
```
images/
  img1.jpg
  img2.jpg
ir/
  img1.jpg
  img2.jpg
labels/
  img1.txt
  img2.txt
```

If I am not mistaken, having a folder structure like this should allow you to simply run the `python train.py`-program ca. as instructed in tutorial.ipynb (or train.ipynb). 

Is this information correct, @ingin97 ?

**Comment by fyang064** [comment]
Hi Sebastian, thank you so much for the reply! It seems like a issue in the dataloader when I tried to load the ir images. I just don't know how to extract ir information from the train_loader iterator:
![image](https://user-images.githubusercontent.com/44177668/184715440-b47670ee-4f99-4439-a841-246787e9cf96.png)
I have already followed the instructions on preparing the folder structure.

Thanks again!

**Comment by ingin97** [comment]
@fyang064 Because it is ignoring the expected fifth value anyways, you could try to just remove it. Though this might lead to another problem as it might not be the original problem anyways.

line 301 in train.py:
```python
for i, (imgs, irs, targets, paths) in pbar:
```

**Comment by fyang064** [comment]
Thank you for the comments and I have solved the problem. 
It seems like the branch yolo_four_input is not available anymore

**Comment by sebastianvitterso** [comment]
Good that you solved your problem. 

As a side note, the branch is called `four_input`. You can see it on the repository's "home page", under `branches`:

![image](https://user-images.githubusercontent.com/37065184/185051092-849cc0be-9915-4016-8124-c4db5ea670f5.png)

**Comment by Yousef-Alhattab** [comment]
> Hi Sebastian, thank you so much for the reply! It seems like a issue in the dataloader when I tried to load the ir images. I just don't know how to extract ir information from the train_loader iterator: ![image](https://user-images.githubusercontent.com/44177668/184715440-b47670ee-4f99-4439-a841-246787e9cf96.png) I have already followed the instructions on preparing the folder structure.
> 
> Thanks again!

@fyang064 could you please share what you have done to solve it?
thanks
